### PR TITLE
On windows set core.longpaths true

### LINKED ISSUE #8744: Every pull request fails with 500
- Gitea version (or commit ref): at least 1.8 to 1.9.4
- Git version:  2.23.0.windows.1
- Operating system: windows
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:
git merge --no-ff --no-commit [C:/Gitea/data/tmp/local-repo/merge-155965200.git]: exec(10:PullRequest.Merge (git merge --no-ff --no-commit): C:/Gitea/data/tmp/local-repo/merge-155965200.git) failed: exit status 128(<nil>) stdout:  stderr: error: Cannot update sparse checkout: the following entries are not up to date:

### PR DESCRIPTION
**Pull request by zeripath**
Fix #8744 

#8744 is reveals a problem with long filenames on windows due to the default length of filenames being limited to 260 characters. 

This PR sets the core.longpaths argument if we are running on windows to allow git to avoid this problem. 

(If in future we can switch merges to using temporary files instead of actual checks out we will likely be able to remove this option.)

### CODE REVIEW

File: modules/git/git.go [hunk 1]
```diff
@@ -133,6 +134,13 @@ func Init() error {
 			return fmt.Errorf("Failed to execute 'git config --global gc.writeCommitGraph true': %s", stderr)
 		}
 	}
+
+	if runtime.GOOS == "windows" {
+		if _, stderr, err := process.GetManager().Exec("git.Init(git config --global core.longpaths true)",
```
Comment (guillep2k) [inline modules/git/git.go:139]: I've tried to find good documentation about this flag and what are the implications of setting it to true. The best I could find is [this link](https://www.brycevandyk.com/of-too-long-file-names-in-windows-and-git/), which doesn't talk about that but does a good explanation about where the feature comes from.

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
This is something to look for if we move to gogit or other similar library.

File: modules/git/git.go (hunk 1 above)
Comment (zeripath) [reply modules/git/git.go:139]: I too searched and it doesn't appear to be in the git source code.I suspect that this is something that they pass to mingw.

Just read your link and I see why I couldn't find it.

**Comment by zeripath** [comment]
As I write here https://github.com/go-gitea/gitea/issues/8744#issuecomment-548969431 the answer is likely to stop checking out arbitrary file paths and make merge work on temp files which we add to the index.

This is how every other update to repos work.

The issue is that merge is a bit of a complex thing and rebase similarly - both have been written expecting a work tree although they could do their magic with temp files.

We'd have to rewrite their functionality and although I've been slowly learning how they work I stopped to sit that bloody exam and now have to restart from scratch.

(Diff also needs the plumbing treatment - we're being extremely inefficient there so that probably should be next focus rather than merge.)

**Review by lunny — APPROVED** [review]

**Pull request merged**